src/PostgREST/DbStructure/Proc.hs

```diff
@@ -77,9 +77,13 @@ type ProcsMap = M.HashMap QualifiedIdentifier [ProcDescription]
   An overloaded function can have a different volatility or even a different return type.
   Ideally, handling overloaded functions should be left to pg itself. But we need to know certain proc attributes in advance.
 -}
-findProc :: QualifiedIdentifier -> S.Set Text -> Bool -> ProcsMap -> Maybe ProcDescription
-findProc qi payloadKeys paramsAsSingleObject allProcs = bestMatch
+findProc :: QualifiedIdentifier -> S.Set Text -> Bool -> ProcsMap -> (ProcDescription, Bool)
+findProc qi payloadKeys paramsAsSingleObject allProcs = (fromMaybe fallback bestMatch, isFoundInSchema)
   where
+    -- instead of passing Maybe ProcDescription around, we create a fallback description here when we can't find a matching function
+    -- args is empty, but because "specifiedProcArgs" will fill the missing arguments with default type text, this is not a problem
+    fallback = ProcDescription (qiSchema qi) (qiName qi) Nothing mempty (SetOf $ Composite $ QualifiedIdentifier mempty "record") Volatile False
```

Comment:
Hm, I was liking how the fallback went away before; because we no longer fallback and instead err right?

Would it possible to remove the fallback and still give a clean error?

Reply:
I think It may be possible, although there are some error validating functions that use `target` with that fallback value, like this one

https://github.com/laurenceisla/postgrest/blob/cd6d6784c2ee34cd5b987f162236cfbe8d929919/src/PostgREST/Request/ApiRequest.hs#L168

I'll check and let you know if it breaks too many things.